#629 - cleanup Checkbox component

TheHadiAhmadi opened the pull request.

TheHadiAhmadi commented:
@pournasserian this is ready to review

pournasserian commented:
@TheHadiAhmadi examples are not correct. What is the usage of description? Where is the SCSS files starting with _?

TheHadiAhmadi commented:
@pournasserian I updated examples and renamed scss files.

and here is an example of checkbox description
![image](https://user-images.githubusercontent.com/42554876/208176807-195e6c45-9239-4189-963f-a69077a990b8.png)

pournasserian commented:
@TheHadiAhmadi so we will need that for Radio as well?

pournasserian commented:
@TheHadiAhmadi create a new issue for description property

pournasserian commented:
@TheHadiAhmadi Checkbox should has checked property (boolean)
Also Indeterminate boolen property is needed
Please see here: 
https://getbootstrap.com/docs/5.2/forms/checks-radios/#indeterminate

Checkbox, CheckboxGroup, FormCheckboxGroup's functionalities should be the same as Radio, RadioGroup, and FormRadioGroup components except value properties in grouping: value in Radio Groups is singular, but in CheckboxGroup it is an array.

TheHadiAhmadi commented:
> @TheHadiAhmadi so we will need that for Radio as well?

Yes, I think

TheHadiAhmadi commented:
@pournasserian what would be difference between `checked` and `value` in Checkbox and FormCheckbox components?

pournasserian commented:
> @pournasserian what would be difference between `checked` and `value` in Checkbox and FormCheckbox components?

`value ` could be an `object`, `string`, etc., but checked is always `boolean | undefined`

TheHadiAhmadi commented:
I updated the code, and I don't know how to use `value` prop in checkbox, is it same as checked in checkbox? (Not checkboxGroup).

Also implemented indeterminate same as disabled (if a checkbox has truthy indeterminate prop, always will be indeterminate unless we make if falsey).

Is it okay or it's better to be dependent of checked prop:

- [x] checked = true
- [ ] checked = false
- [-] checked = indeterminate

TheHadiAhmadi commented:
@pournasserian other than those two above issues, there is no any problem with this PR (I think).

pournasserian commented:
> I updated the code, and I don't know how to use `value` prop in checkbox, is it same as checked in checkbox? (Not checkboxGroup).
> 
> Also implemented indeterminate same as disabled (if a checkbox has truthy indeterminate prop, always will be indeterminate unless we make if falsey).
> 
> Is it okay? or it's better to be dependent of checked prop:
> 
> * [x]  checked = true
> * [ ]  checked = false
> * [-] checked = undefined

checked and `indeterminate` are `boolean | undefined` properties.

TheHadiAhmadi commented:
Okay now I think this is ready to review again.
